30413/Andreea Alexandra Stan/Lab05/Lab05 Extra/main.c

```diff
@@ -0,0 +1,360 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+//open the files/check if they can be opened
+FILE *openAndCheck(char *filePath, char *mode)
+{
+    FILE *f = fopen(filePath, mode);
+    if (NULL == f)
+    {
+        printf("Cannot open the file");
+        exit(0);
+    }
+    return f;
+}
+
+//functions for the tree:
+
+
+typedef struct node_type
+{
+    char key[7];
+    struct node_type *left, *right;
+
+} BSTNodeT;
+
+
+//functions to check if the word read is a character or a number
+int check_node_number(char s[])
+{
+    if(strchr("0123456789",s[0]))
+        return 1;// is a number
+    if(s[0]=='-' && s[1]!='\0')
+        return 1;
+    return 0;//is not a number
+}
+
+int check_node_operator(char s[])
+{
+    if((s[0]=='+' || s[0]=='-' || s[0]=='*' ||s[0]=='/')&& s[1]=='\0')
+        return 1;// is an operator
+    return 0;//is not an operator
+}
+
+//function to create a node
+BSTNodeT *createNode(char givenKey[])
+{
+
+    BSTNodeT *p = (BSTNodeT *) malloc (sizeof (BSTNodeT));
+    if (p)
+    {
+        strcpy(p->key,givenKey);
+        p->left = p-> right = NULL;
+    }
+    return p;
+}
+
+//function to insert a node:
+BSTNodeT *insertNode(BSTNodeT *root, BSTNodeT *node1, FILE *f_out)
+{
+    /* If the tree is empty, then a single node tree is created */
+    if (root == NULL)
+        root = node1;
+    else
+    {
+        if (root->left==NULL)
+            root->left = node1;
+
+        else
+            root->right = node1;
```

Comment:
Here i tried to add "else if (root->right==NULL)" but it doesn't work properly and I don't understand why.

Reply:
It's because you're not using {}. Please use brackets for all your ifs and fors and whiles.